src/etotal.ado

```diff
@@ -66,10 +72,25 @@ capture program drop etotal
 
 program define etotal
 
-	syntax [anything] [if] [fweight pweight iweight], 	                   ///
-		   [level(cilevel)] [over(varname numeric)] [MATrix(name)]
-    
+	syntax [anything] [if] [fweight pweight iweight], 	                ///
+		   [svy] [level(cilevel)] [over(varname numeric)] 				///
+		   [MATrix(name)] [cformat(string)]
+		   
+		   
     
+	if "`svy'" != "" {
+	    local is_svyset = `"`r(settings)'"' != ", clear"
+		if !`is_svyset' {
+		    display as error "data not set up for svy, use {help svyset}"
+			exit 119
+		}
+		if "`weight'" != "" {
+		    display as error "weights not allowed with {bf:svy}"
+			exit 184
+		}
+		local svy "svy:"
+	}
+
 	* if over is specified, get value labels for results matrix ---------------
 	
 	if "`over'" != "" {
```

Comment:
I'm having some issues with `over` and I'm not sure why. It seems like it could have something to do with the number of levels/labels of a categorical variable. I'm working on a person-level 2018 ACS PUMS file:
```Stata
label_acs_pums, year(2018)
gen all = 1
etotal all [iw = pwgtp], over(sex) // works fine, 2 levels
etotal all [iw = pwgtp], over(rac1p) // returns 'type mismatch' error, 8 levels
gen race = rac1p // labels not carried over
etotal all [iw = pwgtp], over(race) // works
label values race rac1p_lbl
etotal all [iw = pwgtp], over(race) // same error as before
```

Reply:
Ah, I know what this is. It's because you can't set a matrix rowname equal to a string that contains a period, as in "Another Race of Mult. Races, not Latino" because Stata interprets "." as factor variable operator. So I will add a ustrregexra() to the first chunk to take all the periods out. I will also truncate at 32 characters while I'm at it, since that's the limit for matrix rownames.

Reply:
Also, you do not have to generate `all`! If you don't specify a varname or exp, etotal will create a tempvar equal to 1 and total the tempvar.

Reply:
Haha yes yes, force of habit!

Reply:
I had not read your reprex carefully enough. The value labels of `rac1p` don't contain a period (I was thinking of `generate_race_var`'s value labels), so that wasn't the problem. It was that `over_lbls` needed to be compound double quoted on line 244. So, I guess at least I ended up fixing two problems instead of one?